### ISSUE TITLE
Curve_parameters.json 

### ISSUE THREAD

**Issue by jiaofang1222** [wvangansbeke/LaneDetection_End2End#36]
Hello, I am experimenting with your model, but the result  is not good, I guess there is a problem with the curve parameters that I generated from my own dataset. So I first use the M-inv in your code to make a perspective transformation of the TuSimple data and then get the curve parameters, but unlike the file Curve_parameters.json, can you tell me how to generate the details of the curve parameters?

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,

It is indeed very important to compute the homography correctly so that it is valid for more or less all images. This homography is fixed in our case, but it is different for every dataset. You need to find the relationship between 4 points in the source image and destination image to compute a specific `M`. I did it by taking 4 on the lane lines for the egolane. Take a look where I compute the homography matrix. You can find the source points and the destination points [here](https://github.com/wvangansbeke/LaneDetection_End2End/blob/5f3bf44a57a4e020b6829c9989104dd2178a74b5/Dataloader/Load_Data_new.py#L301). 

To compute the curve parameters: 1. Get your x, y pixel coordinates, 2. Do the homogeneous transformation, 3. transform back and normalize, 4. Do the fitting.
I will maybe put some code online, since I get a lot of questions about the exact computations.

Kind regards,
Wouter

**Comment by jiaofang1222** [comment]
Thank you for your reply.
I get the coordinates of each lane line by np.where() in the label in ground_truth. Is this correct?

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,

It depends. Try visualizing it. Then you can be sure about it.

Kind regards,
Wouter

**Comment by jiaofang1222** [comment]
Thank you for your reply. I will try to visualize it.

**Comment by jiaofang1222** [comment]
Hello, I have done the curve parameter fitting according to your method. After visualization, I found that the acquisition point is OK, and the loss in the training process is also small, but the lane line fitting effect is very poor. Can you give me some ideas?

**Comment by jiaofang1222** [comment]
Average AREA**2-loss on training set is 0.16554130
===> Average AREA**2-loss on validation set is 0.16511871
===> Average exact area on training set is 0.22435421
===> Average exact area on validation set is 0.22399652

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,

If you are trying to extract 4 lane lines, maybe you can try first training on the egolane lines. In the second stage you can add all 4. Another possibility is to add a segmentation proxy branch. That should definitely help. 

Did you adopt the same ordering convention in the ground truth? The ground truth is ordered relative to the car here.

Kind regards,
Wouter
